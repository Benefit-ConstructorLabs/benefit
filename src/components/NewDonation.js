import React from 'react';
import DonationHeaderContainer from '../containers/DonationHeaderContainer';
import DonationFormContainer from '../containers/DonationFormContainer';
import RecipientProfile from './RecipientProfile';
import DonationPayment from './DonationPayment';

const NewDonation = ({ match }) => (
  <React.Fragment>
    <DonationHeaderContainer match={match} />
    <DonationFormContainer />
    <RecipientProfile />
    <DonationPayment />
  </React.Fragment>
);

export default NewDonation;
