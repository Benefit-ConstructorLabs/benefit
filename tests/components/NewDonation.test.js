import React from 'react'
import NewDonation from '../../src/components/NewDonation';
import renderer from 'react-test-renderer';

describe('NewDonation', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<NewDonation />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})