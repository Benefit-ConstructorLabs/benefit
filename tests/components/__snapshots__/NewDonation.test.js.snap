// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDonation should render correctly 1`] = `
Array [
  <div
    className="donation"
  >
    <h2
      className="donation__headline"
    >
      Cashless donations that make a difference
    </h2>
    <p
      className="donation__strapline"
    >
      Quick. Secure. Cashless.
    </p>
    <ol
      className="donation__steps"
    >
      <li
        className="donation__step"
      >
        Decide how much you would like to donate to firstname.
      </li>
      <li
        className="donation__step"
      >
        Set your payment amount.
      </li>
      <li
        className="donation__step"
      >
        Help firsname out.
      </li>
    </ol>
  </div>,
  <div
    className="donation"
  >
    <img
      alt="Recipient photo"
      className="donation__recipient__photo"
      src="https://via.placeholder.com/150?text=Recipient+photo"
    />
    <h3
      className="donation__purpose"
    >
      Donate towards a night in a shelter
    </h3>
    <form
      className="donation__form"
      onSubmit={[Function]}
    >
      <h4>
        How much?
      </h4>
      <p>
        <input
          className="donation__form__amount"
          id="donation__amount"
          placeholder="£1"
          type="number"
        />
      </p>
      <p>
        <input
          type="range"
        />
      </p>
      <p
        className="donation__form__quick-amount"
      >
        <button
          className="btn btn__quick-amount"
        >
          50p
        </button>
        <button
          className="btn btn__quick-amount"
        >
          £1
        </button>
        <button
          className="btn btn__quick-amount"
        >
          £2
        </button>
      </p>
      <button
        className="btn btn__submit"
        type="submit"
      >
        Donate to firstname
      </button>
    </form>
  </div>,
  <div
    className="recipient-profile"
  >
    <h3
      className="recipient-profile__headline"
    >
      Fancy a chat?
    </h3>
    <p
      className="recipient-profile__intro"
    >
      firstName is interested in:
    </p>
    <ol
      className="recipient-profile__interests"
    >
      <li
        className="recipient-profile__interest"
      >
        First interest
      </li>
      <li
        className="recipient-profile__interest"
      >
        Second interest
      </li>
      <li
        className="recipient-profile__interest"
      >
        Third interest
      </li>
    </ol>
  </div>,
  <h2>
    Donation Payment
  </h2>,
  <div
    className="payment-details-form"
  >
    <h3>
      Set up payment
    </h3>
    <form
      className="payment-details-form__form"
    >
      <input
        placeholder="Card number"
        type="text"
      />
      <input
        placeholder="Expiry date"
        type="text"
      />
      <input
        placeholder="CCV"
        type="text"
      />
      <p>
        Set up a donor account?
      </p>
      <p>
        Continue donating anonymously:
      </p>
      <button>
        Donate -Price- to -First Name-
      </button>
    </form>
  </div>,
  <section
    className="acknowledgement"
  >
    <p
      className="acknowledgement__notification"
    >
      Donation confirmed
    </p>
    <h2
      className="acknowledgement__confirmation"
    >
      All set. Thanks for helping "FirstName" out!
    </h2>
    <h3
      className="acknowledgement__question"
    >
      Want to save your details for another donation?
    </h3>
    <button
      className="btn acknowledgement__account__btn"
      onClick={[Function]}
    >
      Set up donor account
    </button>
  </section>,
]
`;
